### PR TITLE
update msquic

### PR DESCRIPTION
**Pull request by wfurt**
This updates Quic project to consume our binaries by default for windows build. The dotnet/msquic was updated to produce upcoming 2.2 version (built from main)
I also updated version of official msquic binaries. There don't seems to be any critical fixes but it feels right to point it at latest release.

### CODE REVIEW

**Review by rzikm — APPROVED** [review]
LGTM!

File: eng/Versions.props [hunk 1]
```diff
@@ -221,8 +221,8 @@
     <!-- ICU -->
     <MicrosoftNETCoreRuntimeICUTransportVersion>8.0.0-preview.2.23103.1</MicrosoftNETCoreRuntimeICUTransportVersion>
     <!-- MsQuic -->
-    <MicrosoftNativeQuicMsQuicVersion>2.1.1</MicrosoftNativeQuicMsQuicVersion>
-    <SystemNetMsQuicTransportVersion>7.0.0-alpha.1.22406.1</SystemNetMsQuicTransportVersion>
+    <MicrosoftNativeQuicMsQuicVersion>2.1.7</MicrosoftNativeQuicMsQuicVersion>
+    <SystemNetMsQuicTransportVersion>8.0.0-alpha.1.23106.6</SystemNetMsQuicTransportVersion>
```
Comment (ManickaP) [inline eng/Versions.props:225]: Does maestro not work anymore for this? I thought this should be flown automatically?
If we now update dotnet/msquic, will it get automatically propagated?

File: src/libraries/System.Net.Quic/src/System.Net.Quic.csproj [hunk 2]
```diff
@@ -12,7 +12,9 @@
     <DefineConstants Condition="'$(TargetPlatformIdentifier)' == 'windows'">$(DefineConstants);TARGET_WINDOWS</DefineConstants>
     <GeneratePlatformNotSupportedAssemblyMessage Condition="'$(TargetPlatformIdentifier)' == ''">SR.SystemNetQuic_PlatformNotSupported</GeneratePlatformNotSupportedAssemblyMessage>
     <ApiExclusionListPath Condition="'$(TargetPlatformIdentifier)' == ''">ExcludeApiList.PNSE.txt</ApiExclusionListPath>
-    <UseQuicTransportPackage  Condition="'$(UseQuicTransportPackage)' == ''">false</UseQuicTransportPackage>
+    <!-- This controls if we consume official binaries from MsQuic or if we use binaries published from dotnet/msquic repo.
```
Comment (ManickaP) [inline src/libraries/System.Net.Quic/src/System.Net.Quic.csproj:15]:
```suggestion
    <!-- This controls if we consume official MsQuic binaries for Windows from NuGet (false) or if we use binaries published from dotnet/msquic repo (true).
```

File: src/libraries/System.Net.Quic/src/System.Net.Quic.csproj [hunk 3]
```diff
@@ -139,7 +141,7 @@
     <Reference Include="System.Diagnostics.StackTrace" Condition="'$(Configuration)' == 'Debug'" />
   </ItemGroup>
 
-  <!-- Support for deploying msquic on Windows -->
+  <!-- Support for deploying msquic on Windows. We make msquic.dll part or runtime binaries. (either from official releases or our transport) -->
```
Comment (ManickaP) [inline src/libraries/System.Net.Quic/src/System.Net.Quic.csproj:144]:
```suggestion
  <!-- Support for deploying msquic on Windows. We make msquic.dll part of runtime binaries: either from official releases or our transport feed. -->
```

File: eng/Versions.props (hunk 1 above)
Comment (wfurt) [reply eng/Versions.props:225]: we should check. But I did not want to wait for the cycle and it does not update MicrosoftNativeQuicMsQuicVersion

File: src/libraries/System.Net.Quic/src/System.Net.Quic.csproj (hunk 2 above)
Comment (wfurt) [reply src/libraries/System.Net.Quic/src/System.Net.Quic.csproj:15]: I think the transport is also NuGet - just different feed. I'm happy to make updates that make sense to us.

Comment (ManickaP) [reply src/libraries/System.Net.Quic/src/System.Net.Quic.csproj:15]: From NuGet.org? :smile:

Comment (wfurt) [reply src/libraries/System.Net.Quic/src/System.Net.Quic.csproj:15]: the comment did not say NuGet.org :)
and I'm not sure it matters.

File: eng/Versions.props (hunk 1 above)
Comment (wfurt) [reply eng/Versions.props:224]: it seems like maestro just did it.

**Comment by wfurt** [comment]
windows crash looks suspicious. I will investigate...

**Comment by CarnaViire** [comment]
Any news on this @wfurt ?

**Comment by wfurt** [comment]
no. We may need to pick up more changes from MsQuic so I may wait for that before diving deep.

**Comment by wfurt** [comment]
all the test failures are independent. Let me know if it is ok to move this forward @ManickaP @CarnaViire

**Review by ManickaP — APPROVED** [review]
LGTM, thanks!

**Pull request merged**